src/compactor_state.rs

```diff
@@ -472,6 +522,75 @@ mod tests {
         assert_eq!(merged_l0, expected_merged_l0s);
     }
 
+    #[test]
+    fn test_should_submit_invalid_compaction_wrong_order() {
+        // given:
+        let rt = build_runtime();
+        let (_, _, mut state) = build_test_state(rt.handle());
+
+        let l0_sst = &mut state.db_state().l0.clone();
+        let last_sst = l0_sst.pop_back();
+        l0_sst.push_front(last_sst.unwrap());
+        // when:
+        let result = state.submit_compaction(build_l0_compaction(l0_sst, 0));
+
+        // then:
+        assert!(matches!(result, Err(SlateDBError::InvalidCompaction)));
+    }
+
+    #[test]
+    fn test_should_submit_invalid_compaction_skipped_sst() {
+        // given:
+        let rt = build_runtime();
+        let (_, _, mut state) = build_test_state(rt.handle());
+
+        let l0_sst = &mut state.db_state().l0.clone();
+        let last_sst = l0_sst.pop_back().unwrap();
+        l0_sst.push_front(last_sst);
+        l0_sst.pop_back();
+        // when:
+        let result = state.submit_compaction(build_l0_compaction(l0_sst, 0));
+
+        // then:
+        assert!(matches!(result, Err(SlateDBError::InvalidCompaction)));
+    }
+
+    #[test]
+    fn test_should_submit_invalid_compaction_with_sr() {
+        // given:
+        let rt = build_runtime();
+        let (_, _, mut state) = build_test_state(rt.handle());
+
+        let mut compaction = build_l0_compaction(&state.db_state().l0, 0);
+        compaction.sources.push(SourceId::SortedRun(5));
+        // when:
+        let result = state.submit_compaction(compaction);
```

Comment:
is it possible to test `validate_compaction` directly? Not sure if the file structure is appropriate for structuring the tests that way, it would be good if we can test at `validate_compaction` level for these tests instead of `submit_compaction`.